pennylane_lightning/src/bindings/Bindings.cpp

```diff
@@ -320,10 +320,19 @@ void lightning_class_bindings(py::module &m) {
                  return py::array_t<ParamT>(py::cast(M.probs(wires)));
              })
         .def("expval",
-             [](Measures<PrecisionT> &M, const std::string &operation,
-                const std::vector<size_t> &wires) {
-                 return M.expval(operation, wires);
-             })
+             static_cast<PrecisionT (Measures<PrecisionT>::*)(
+                 const std::string &, const std::vector<size_t> &)>(
+                 &Measures<PrecisionT>::expval),
+             "Expected value of an operation by name.")
+#ifdef _ENABLE_KOKKOS
+        .def("expval",
+             static_cast<PrecisionT (Measures<PrecisionT>::*)(
+                 const std::vector<Util::index_type> &,
+                 const std::vector<Util::index_type> &,
+                 const std::vector<std::complex<PrecisionT>> &)>(
+                 &Measures<PrecisionT>::expval),
```

Comment:
I think a performance issue may be from here (and one in `Measures.hpp` file). If we use `std::vector`, pybind always copies the data from Python to internal storage (as `std::vector` should maintain the data). Probably, changing the arguments here to `py::array_t` and using pointers in real implementation.

Reply:
You may check [this](https://github.com/chaeyeunpark/UnionFind/blob/1002d0bf2ed2139792b9bad39bc0496dd0e497b7/UnionFindPy/cpp/binding/UnionFindPy.cpp#L62-L87) in my personal project that handles a sparse array with pybind.

Reply:
Thank you @chaeyeunpark! I will check!

Reply:
@chaeyeunpark, could you please clarify what you mean with "one in Measures.hpp file"?

Reply:
Hi Amintor, I left a comment in a function that may need to be changed.